File: marketplace/adoptium-marketplace-schema/src/main/java/net/adoptium/marketplace/schema/SourcePackage.java
package net.adoptium.marketplace.schema;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

public class SourcePackage {

    @Schema(example = "OpenJDK8U-sources_8u232b09.tar.gz", required = true)
    private final String name;

    @Schema(example = "https://github.com/AdoptOpenJDK/openjdk8-upstream-binaries/releases/download/jdk8u232-b09/OpenJDK8U-sources_8u232b09.tar.gz", required = true)
    private final String link;

    @JsonCreator
    public SourcePackage(
            @JsonProperty("name") String name,
            @JsonProperty("link") String link) {
        this.name = name;
        this.link = link;
    }

    public String getName() {
        return name;
    }

    public String getLink() {
        return link;
    }
}

File: adoptium-models-parent/adoptium-api-v3-models/src/main/kotlin/net/adoptium/api/v3/models/SourcePackage.kt
package net.adoptium.api.v3.models

import org.eclipse.microprofile.openapi.annotations.media.Schema

class SourcePackage(
    name: String,
    link: String,
    size: Long) : FileAsset(name, link, size)

class ReleaseNotesPackage(
    name: String,
    link: String,
    size: Long) : FileAsset(name, link, size)

open class FileAsset {

    @Schema(example = "OpenJDK8U-sources_8u232b09.tar.gz")
    val name: String

    @Schema(example = "https://github.com/AdoptOpenJDK/openjdk8-upstream-binaries/releases/download/jdk8u232-b09/OpenJDK8U-sources_8u232b09.tar.gz")
    val link: String

    @Schema(example = "82573385")
    val size: Long

    constructor(
        name: String,
        link: String,
        size: Long
    ) {
        this.name = name
        this.link = link
        this.size = size
    }

    override fun toString(): String {
        return "FileAsset(name='$name', link='$link', size=$size)"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as FileAsset

        if (name != other.name) return false
        if (link != other.link) return false
        if (size != other.size) return false

        return true
    }

    override fun hashCode(): Int {
        var result = name.hashCode()
        result = 31 * result + link.hashCode()
        result = 31 * result + size.hashCode()
        return result
    }
}
